frontend/src/types/errorResponse.ts

```diff
@@ -0,0 +1,95 @@
+/**
+ * An error response returned by the API. This implements the error responses
+ * ADR, please refer to doc/adr/0012-error-responses.md for more information.
+ */
+export type ErrorResponse<
+  T extends keyof ErrorResponseDetail = "__fallback__",
```

Comment:
The default to `__fallback__` is needed so we can use `ErrorResponse` without a generic param in places where we don't care about the specific type of error (e.g. in the mapper).